packages/@aws-cdk/aws-rds/lib/cluster.ts

```diff
@@ -744,10 +743,13 @@ export class DatabaseClusterFromSnapshot extends DatabaseClusterNew {
       this.secret = secret.attach(this);
     }
 
-    if (deprecatedCredentials.secret) {
-      const deprecatedSecret = deprecatedCredentials.secret.attach(this);
-      if (!this.secret) {
-        this.secret = deprecatedSecret;
+    if (!props.snapshotCredentials) {
```

Comment:
Isn't this a breaking change because this condition is always `true` for user not using the new prop, so the previous default behavior disappears?

Reply:
(sorry i was out of office) Ohhhhhkay I see. Yes, this would be a breaking change for those cases. Hmm, we *could* add another prop for this specific scenario, or alternatively, hide this breaking change behind a feature flag. WDYT @jogold ?

Reply:
Maybe a feature flag that makes `credentials` a no-op?

Reply:
For people coming here that don't want 2 secrets, use `snapshotCredentials` and then specify `credentials` like this:

```ts
credentials: Credentials.fromPassword('dummy', SecretValue.unsafePlainText('dummy'))
```